props/saber_fett263_G_buttons.h

```diff
@@ -0,0 +1,445 @@
+// or hold AUX and Swing while blade is ON to toggle mode
+//
+// Ignite (ON) - click PWR while OFF or Swing Saber while OFF
+//
+// Muted Ignition (ON) - double click PWR while OFF
+// Retract (OFF) - click PWR while ON (not swinging or in Battle Mode)
+// Twist Off (OFF) if in Battle Mode
+// Play Music Track - hold and release PWR while OFF
+// Blast - click AUX while ON
+// Multi-Blast Mode - hold and release AUX while ON to enter mode, Swing to initiate Blasts, click Aux to exit mode
+// Clash - clash blade while ON
+// Stab - thrust forward and clash blade while ON
+// Lockup - hold AUX and clash while ON
+// Drag - hold AUX and clash pointing down while ON
+// Lightning Block - hold PWR and click AUX while ON
+// Melt - hold PWR (or AUX) and thrust forward and clash while ON
+// Force - hold and release PWR while ON
+// Color Change - hold AUX and click PWR while ON to enter CCWheel,
+// turn hilt to rotate through colors, click PWR to select/exit
+// if using COLOR_CHANGE_DIRECT each button press advances one Color at a time
+// Next Preset - click AUX while OFF
+// MULTI_PHASE Next Preset - hold AUX and TWIST while ON (use define to enable)
+// Previous Preset - hold AUX and click PWR while OFF
+// MULTI_PHASE Previous Preset - hold PWR and TWIST while ON (use define to enable)
+// Enter SA22C Volume Menu - hold and release AUX while OFF
+// Volume Up (10% increment, 100% max) - click PWR while in Volume Menu while OFF
+// Volume Down (10% increment) - click AUX while in Volume Menu while OFF
+// Exit Volume Menu - hold and release AUX while in Volume Menu while OFF
+
+#ifndef MOTION_TIMEOUT
+#define MOTION_TIMEOUT 60 * 15 * 1000
+#endif
+
+#ifndef LOCKUP_THRESHOLD_G
+#define LOCKUP_THRESHOLD_G 2.0
+#endif 
+
+#ifndef PROPS_SABER_FETT263_G_BUTTONS_H
+#define PROPS_SABER_FETT263_G_BUTTONS_H
+
+#include "prop_base.h"
+#include "../sound/hybrid_font.h"
+
+#undef PROP_TYPE
+#define PROP_TYPE SaberFett263Buttons
+
+// The Saber class implements the basic states and actions
+// for the saber.
+class SaberFett263Buttons : public PropBase {
+public:
+SaberFett263Buttons() : PropBase() {}
+  const char* name() override { return "SaberFett263Buttons"; }
+
+
+  // EVENT_SWING
+  bool swinging_ = false;
+  void Loop() override {
+    PropBase::Loop();
+    if(!SaberBase::IsOn() ) {
+      if (millis() - saber_off_time_ < MOTION_TIMEOUT) {
+        SaberBase::RequestMotion();
+        if (!swinging_ && fusor.swing_speed() > 250) {
+            swinging_ = true;
+            Event(BUTTON_NONE, EVENT_SWING);
+            }
+            if (swinging_ && fusor.swing_speed() < 100) {
+            swinging_ = false;
+            }
+      }
+    }
+    if (!swinging_ && fusor.swing_speed() > 250) {
+      swinging_ = true;
+      Event(BUTTON_NONE, EVENT_SWING);
+    }
+    if (!swinging_ && fusor.swing_speed() > 120 && fusor.swing_speed() < 250 && millis() > clash_swing_ + (100 * LOCKUP_THRESHOLD_G)) {
```

Comment:
This seems really weird:  millis() > clash_swing_ + (100 * LOCKUP_THRESHOLD_G)
What it seems to do is to put a maximum length on the lockup?
But why use a define called LOCKUP_THRESHOLD_G for that? The _G indicates that the unit is earth gravities, not deciseconds.  (And why are we using deciseconds?)

Reply:
This allows "fine tuning" for the physical bounce clashing two sabers creates, I misunderstood the "G" from the CLASH_THRESHOLD_G, I have changed to LOCKUP_THRESHOLD_DELAY.  This delay gives the user time to steady their saber to keep lockup active before the end lockup begins.  After the actual clash this delay "ignores" the pull away that gets created if the sabers bounce off object.  Depending on the hilt and user they will need to be able to adjust this delay.

Reply:
LOCKUP_THRESHOLD_DELAY is better, but why should it be specifies the delay in deciseconds? It should either be in seconds or in milliseconds I think.

Reply:
changed to millis